core/crt-core/src/main/java/software/amazon/awssdk/crtcore/CrtConfigurationUtils.java

```diff
@@ -54,11 +60,19 @@ public static Optional<HttpProxyOptions> resolveProxy(CrtProxyConfiguration prox
         return Optional.of(clientProxyOptions);
     }
 
+    private static boolean doesTargetMatchNonProxyHosts(String target, Set<String> hostPatterns) {
```

Comment:
should we extract this same logic that is used by multiple HTTP client into a common utils class?